src/test/java/org/cactoos/iterable/MappedWithIndexTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017-2020 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.iterable;
+
+import java.util.Collections;
+import org.cactoos.text.Joined;
+import org.cactoos.text.TextOf;
+import org.cactoos.text.Upper;
+import org.hamcrest.Matchers;
+import org.hamcrest.core.IsEqual;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.llorllale.cactoos.matchers.Assertion;
+
+/**
+ * Tests for {@link MappedWithIndex}.
+ * @since 1.0.0
+ * @checkstyle MagicNumberCheck (500 lines)
+ */
+final class MappedWithIndexTest {
+    @Test
+    void transformsList() throws Exception {
+        new Assertion<>(
+            "must transform an iterable",
+            new MappedWithIndex<>(
+                (input, index) -> new Upper(
+                    new Joined(
+                        "-",
+                        new TextOf(index.toString()),
+                        new TextOf(input)
+                    )
+                ),
+                new IterableOf<>(
+                    "hello", "world", "друг"
+                )
+            ).iterator().next().asString(),
+            new IsEqual<>("0-HELLO")
```

Comment:
@dearrudam here I believe it's best to simply test equality to the fully built `Iterable`, not just one element, also let's try to keep tests as simple as needed to demonstrate that the object work: here I believe we don't need to combine upper + joined, some simpler string concatenate (with `Joined` or `Concatenated`) should be enough.

Reply:
@victornoel I'll fix it! Thanks!